cmd/daemon.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/takama/daemon"
+)
+
+var d daemon.Daemon
+var InstallPort string
+
+var daemonCmd = &cobra.Command{
+	Use:   "daemon",
+	Short: "Start the Showcase server as a daemon",
+	Long: `This command handles the Showcase server in a daemon. The daemon has a
+life cycle in which the daemon must be installed using the install subcommand,
+then started, and finally stopped.`,
+}
+
+var installCmd = &cobra.Command{
+	Use:   "install",
+	Short: "Install the Showcase server daemon",
+	Run: func(cmd *cobra.Command, args []string) {
+		status, err := d.Install("start", "--port", InstallPort)
+		if err != nil {
+			ErrLog.Println(status, "\nError: ", err)
+			os.Exit(1)
+		}
+		StdLog.Println(status)
+	},
+}
+
+var daemonStartCmd = &cobra.Command{
```

Comment:
I'm not very familiar with the daemon. What's the difference between installing and starting?

Reply:
For linux:

`Install` creates a shell script that is placed at [/etc/init.d/](https://askubuntu.com/questions/5039/what-is-the-difference-between-etc-init-and-etc-init-d) I assume this is the common place to put daemon-like executables. 

`Start` calls the shell script to invoke the go program.

For other platforms, the commands do behaviorally similar things but with what is idiomatic for the platform.

Reply:
Ah